docs/DeveloperGuide.md

```diff
@@ -0,0 +1,123 @@
+# RepoSense - Developer Guide
+Thank you for your interest in contributing to RepoSense!
+- [Setting up](#setting-up)
+  - [Prerequisites](#prerequisites)
+  - [Setting up the project in your computer using IntelliJ](#setting-up-the-project-in-your-computer-using-intellij)
+  - [Verifying the setup](#verifying-the-setup)
+  - [Configuring the coding style](#configuring-the-coding-style)
+- [Architecture](#architecture)
+  - [Parser](#parser)
+  - [Git](#git)
+  - [CommitsReporter](#commitsreporter)
+  - [AuthorshipReporter](#authorshipreporter)
+  - [ReportGenerator](#reportgenerator)
+  - [System](#system)
+  - [Model](#model)
+- [HTML Dashboard](#html-dashboard)
+
+## Setting up
+
+### Prerequisites
+1. **JDK `1.8.0_60`** or later.
+2. **git** on the command line.
+3. **findstr** for *Windows*, **grep** for *macOS* or *Linux* on the command line.
+ > Check that the tools exist on your OS terminal by typing its name on the terminal and ensure that the terminal does not output messages such as `not found` or `not recognized`.
+
+### Setting up the project in your computer using IntelliJ
+1. Fork this repo, and clone the fork to your computer.
+2. Open *IntelliJ* (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project dialog first).
+3. Set up the correct *JDK* version for *Gradle*.
+4. Click `Configure` > `Project Defaults` > `Project Structure`.
+5. Click `New…​` and find the directory of the *JDK*.
+6. Click `Import Project`.
+7. Locate the `build.gradle` file and select it. Click `OK`.
+8. Ensure that the selected version of `Gradle JVM` matches our prerequisite.
+9. Click `OK` to accept the all the other default settings.
+
+### Verifying the setup
+1. Ensure that *Gradle* build without error.
+2. Run the tests to ensure they all pass.
+   1. On the project root directory, run the command `gradlew clean build functional` and ensure the build is successful. Note that the `clean build` and `functional` command may be required to be run separately.
```

Comment:
This is a bit vague as there is no explanation/guidance given as to why and under what conditions they need to run separately. May be just ask the two to be run separately?